tests/dialects/onnx/test_broadcast.py

```diff
@@ -0,0 +1,76 @@
+import pytest
+
+from xdsl.dialects.onnx import (
+    multidirectional_broadcast_shape,
+    unidirectional_broadcast_shape,
+)
+
+m_test_cases: list[tuple[tuple[list[int], list[int]], list[int]]] = [
+    (([2, 3, 4, 5], []), [2, 3, 4, 5]),
+    (([2, 3, 4, 5], [5]), [2, 3, 4, 5]),
+    (([4, 5], [2, 3, 4, 5]), [2, 3, 4, 5]),
+    (([1, 4, 5], [2, 3, 1, 5]), [2, 3, 4, 5]),
+    (([3, 4, 5], [2, 1, 1, 1]), [2, 3, 4, 5]),
+]
+
+u_test_cases: list[tuple[tuple[list[int], list[int]], list[int]]] = [
+    (([2, 3, 4, 5], []), [2, 3, 4, 5]),
+    (([2, 3, 4, 5], [5]), [2, 3, 4, 5]),
+    (([2, 3, 4, 5], [2, 1, 1, 5]), [2, 3, 4, 5]),
+    (([2, 3, 4, 5], [1, 3, 1, 5]), [2, 3, 4, 5]),
+]
+
+m_fail_test_cases: list[tuple[tuple[list[int], list[int]], list[int]]] = [
+    (([2, 3, 4, 5], []), []),
+    (([2, 3, 4, 5], [5]), [2, 3, 4, 5]),
+    (([4, 5], [2, 3, 4, 5]), [2, 3, 4, 5]),
+    (([1, 4, 5], [2, 3, 1, 5]), [2, 4, 5, 5]),
+    (([3, 4, 5], [2, 1, 1, 1]), [3, 3, 4, 5]),
+]
```

Comment:
Could you please add these inputs to the test cases above with `None`? It's pretty much always better to unit test for expected result, as opposed to `!=`